### PR TITLE
[TACHYON-1673] Add -R option in ls/rm and deprecate lsr/rmr

### PR DESCRIPTION
**Pull request by ifcharming**
https://tachyon.atlassian.net/browse/TACHYON-1673

This is the last PR to address TACHYON-1673, it
- refactors runCommand() method to take CommandLine option
- adds -R support in ls/rm
- makes lsr/rmr as alias, deprecated.
- removes LsrCommand, RmrCommand and AbstractLsCommand classes.

### CODE REVIEW

File: shell/src/main/java/tachyon/shell/command/RmCommand.java [hunk 1]
```diff
@@ -44,16 +49,38 @@ public String getCommandName() {
   }
 
   @Override
-  void runCommand(TachyonURI path) throws IOException {
+  protected int getNumOfArgs() {
+    return 1;
+  }
+
+  @Override
+  protected Options getOptions() {
```
Comment (aaudiber) [inline shell/src/main/java/tachyon/shell/command/RmCommand.java]: We could save a bunch of boilerplate by defining the recursive option as a protected constant in `AbstractTfsShellCommand`, and having this entire function body be 
`return new Options().addOption(RECURSIVE_OPTION)`

File: shell/src/main/java/tachyon/shell/command/RmCommand.java [hunk 2]
```diff
@@ -44,16 +49,38 @@ public String getCommandName() {
...
         throw new IOException("Path " + path + " does not exist");
       }
-      if (mFileSystem.getStatus(path).isFolder()) {
-        throw new IOException("rm: cannot remove a directory, please try rmr <path>");
+      if (!recursively && mFileSystem.getStatus(path).isFolder()) {
+        throw new IOException("rm: cannot remove a directory, please try rm -R <path>");
       }
-      mFileSystem.delete(path);
+
+      DeleteOptions options = DeleteOptions.defaults().setRecursive(recursively);
```
Comment (aaudiber) [inline shell/src/main/java/tachyon/shell/command/RmCommand.java]: it's called `setRecursive`, so lets call it `recursive` everywhere

File: shell/src/main/java/tachyon/shell/command/RmCommand.java [hunk 3]
```diff
@@ -62,11 +89,11 @@ void runCommand(TachyonURI path) throws IOException {
...
   @Override
   public String getUsage() {
-    return "rm <path>";
+    return "rm [-R] <path>";
   }
 
   @Override
   public String getDescription() {
-    return "Removes the specified file.";
+    return "Removes the specified file.  Specify -R to remove file or directory recursively.";
```
Comment (aaudiber) [inline shell/src/main/java/tachyon/shell/command/RmCommand.java]: we typically use only one space after periods

**Comment by aaudiber** [comment]
Thanks for the improvement, I've left a few comments

File: shell/src/main/java/tachyon/shell/command/RmCommand.java (hunk 1 above)
Comment (ifcharming) [inline shell/src/main/java/tachyon/shell/command/RmCommand.java]: Done, thanks!

File: shell/src/main/java/tachyon/shell/command/RmCommand.java (hunk 2 above)
Comment (ifcharming) [inline shell/src/main/java/tachyon/shell/command/RmCommand.java]: Done.

File: shell/src/main/java/tachyon/shell/command/RmCommand.java (hunk 3 above)
Comment (ifcharming) [inline shell/src/main/java/tachyon/shell/command/RmCommand.java]: Updated all.

**Comment by ifcharming** [comment]
Thanks @aaudiber for the review! Comments addressed, PTAL.

**Comment by aaudiber** [comment]
LGTM

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**